### DISCUSSION TITLE
Train data vs dev data for spacy

### DISCUSSION THREAD

**Discussion by DavidGerva** [explosion/spaCy#8395] [Help: Other Questions] [answered]
How do spacy model training uses "train" data and "dev" data? 

In the config.cfg file I can specify the two params `paths.train` and `paths.dev` but I cannot understand how spacy is using the two: when I train my model I don't see any output specifically related to them.
I certainly miss something about this, can you give me some links/material to explore?

**Comment by polm** [comment, marked as answer]
When you train with training and dev data, the training data is actually used for training (to update your model), while the dev data is used for evaluation that occurs every however-many steps. 

The dev data is referred to as "evaluation data" at the start of the [training docs](https://spacy.io/usage/training).

**Reply by DavidGerva** [reply]
Thank you @polm! So, just to be sure I understood correctly, if I don't specify the dev data spacy will use the train data also for evaluation?

**Reply by svlandeg** [reply]
You could easily try this out: if you run `spacy train` with no `dev` corpus, a `ValueError` will be raised.

You could specify the same training set also for the "dev" portion, and then the ML algorithm will be trained AND evaluated on the same corpus. Do note that this is not an ideal scenario though, because you won't know whether or not your algorithm is overfitting, and you won't have a realistic view on the performance of your model.

**Reply by DavidGerva** [reply]
Awesome! Thanks a lot!
